google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -321,50 +348,385 @@ private boolean isAlive() {
 
   @Override
   public void sendAckOperations(
-      List<String> acksToSend, List<PendingModifyAckDeadline> ackDeadlineExtensions) {
-    ApiFutureCallback<Empty> loggingCallback =
-        new ApiFutureCallback<Empty>() {
-          @Override
-          public void onSuccess(Empty empty) {
-            ackOperationsWaiter.incrementPendingCount(-1);
-          }
+      List<MessageDispatcher.AckWithMessageFuture> acksToSend,
+      List<PendingModifyAckDeadline> ackDeadlineExtensions) {
+    Set<String> failedModackIds = sendModacks(ackDeadlineExtensions, new HashSet<String>());
```

Comment:
Why are we sending modack in `sendAckOperations`?

Reply:
`sendAckOperations` is a catch-all for both acks and modacks. This was how the call was setup in the existing codebase, I am open to changing the name to make it more clear.